examples/federated_statistics/df_stats/README.md

```diff
@@ -0,0 +1,314 @@
+
+fed_stats parser
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --prepare-data        prepare data based on configuration
+
+```
+
+```
+python3 data_utils.py  --prepare-data
+
+prepare data for poc workspace:/tmp/nvflare/poc
+remove existing data at /tmp/nvflare/poc/site-1/data.csv
+wget download to /tmp/nvflare/poc/site-1/data.csv
+100% [..........................................................................] 3974305 / 3974305remove existing data at /tmp/nvflare/poc/site-2/data.csv
+wget download to /tmp/nvflare/poc/site-2/data.csv
+100% [..........................................................................] 2003153 / 2003153done with prepare data
+
+```
+If your poc_workspace is in a different location, use the following command
+
+```
+export NVFLARE_POC_WORKSPACE=<new poc workspace location>
+```
+then repeat above
+
+
+## 3. Compute the local and global statistics for each numerical feature
+
+### 3.1 Specify client side configuration
+
+We are using a built-in NVFLARE executor,  
+
+```
+ "executor": {
+        "id": "Executor",
+        "path": "nvflare.app_common.executors.statistics_executor.StatisticsExecutor",
+        "args": {
+          "generator_id": "df_stats_generator",
+          "min_count" : 10,
+          "min_random": 0.1,
+          "max_random": 0.3
+  },
+
+```
+here we specify a minimum of 10 rows (min_count), if the number of rows less than min_count, the job will fail.
+In case, user will need to calculate histogram, user will need to provide the min/max value of given feature. 
+But instead of directly send the local min/max to server, we introduce some randomness to the min/max value. 
+
+so that estimate min = min * ( 1 - random ); max = max * ( 1 + random )
+where random value is within min_random and max_random, in this example ( 0.1 and 0.3) ( data privacy police defined noise level)
+
+The estimated global min and max should be 
+
+estimated global min <  all clients' min value and  global max >  all clients' max value
+and global min < global max
+
+We then use this estimate global min and max to dynamic calculate the histogram range for all histogram buckets ( client and global)
+if the feature range is not specified. 
+
+
+Then user need to implemented the local statistics generator, identified as "generator_id"
+```
+"components": [
+    {
+    "id": "df_stats_generator",
+    "path": "df_stats_generator.DFStatistics",
+    "args": {
+        "data_path" : "{workspace_dir}/{client_name}/data.csv",
+    }
+}
+```
+In this example, Data Frame Statistics generator is DFStatistics and it takes input data_path for each client.  
+
+
+### 3.2 Specify Server side configuration
+ 
+Here we use the built-in Controller, called GlobalStatistics. Here we selected all the available metrics.  
+
+```
+"workflows": [
+    {
+      "id": "fed_stats_controller",
+      "path": "nvflare.app_common.workflows.global_statistics.GlobalStatistics",
```

Comment:
If "Statistics" stay the same name, suggest change this name to "GlobalStatisticsController" or something to avoid confusion